### DISCUSSION TITLE
is all tailwind is on rem ?

### DISCUSSION THREAD

**Discussion by 18121A05L2** [tailwindlabs/tailwindcss#9361] [Help]
--> hey I had a doubt like , we are using all the time p-2 , h-3 , gap-4  -- is all these things comes under the rems or the pixels  ?

--> and also another question is using all these and by making changes to the html which changes in whole dom , will it reflect ?

**Comment by wongjn** [comment]
> we are using all the time p-2 , h-3 , gap-4 -- is all these things comes under the rems or the pixels ?

You can [look in the documentation for the default spacing scale](https://tailwindcss.com/docs/customizing-spacing#default-spacing-scale) that many of the utilities (and all the ones you listed) use. As you'll see from the documentation, only the `px` value uses pixels, whereas the rest use `rem` values. It is important to note some exceptions, such as the [border width](https://tailwindcss.com/docs/border-width) utilities. One can always reference the documentation for the particular utilities group to check if `rem` or pixels are used.

> making changes to the html which changes in whole dom , will it reflect ?

If I am understanding the question correctly, changing the `font-size` on the `<html>` will affect the size of these spacing lengths, as per the [MDN documentation on the `rem` unit](https://developer.mozilla.org/en-US/docs/Web/CSS/length):

> Represents the [`font-size`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-size) of the root element (typically [`<html>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/html)). When used within the root element [`font-size`](https://developer.mozilla.org/en-US/docs/Web/CSS/font-size), it represents its initial value (a common browser default is `16px`, but user-defined preferences may modify this).
